ste/jobStatusManager.go

```diff
@@ -56,19 +61,73 @@ func (jm *jobMgr) SendXferDoneMsg(msg xferDoneMsg) {
 }
 
 func (jm *jobMgr) ListJobSummary() common.ListJobSummaryResponse {
-	jm.jstm.listReq <- true
-	return <-jm.jstm.respChan
+	if !jm.jstm.doneHandleStatusUpdateManager {
+		jm.jstm.listReq <- true
+		return <-jm.jstm.respChan
+	} else {
+		return common.ListJobSummaryResponse{}
+	}
 }
 
 func (jm *jobMgr) ResurrectSummary(js common.ListJobSummaryResponse) {
 	jm.jstm.js = js
 }
 
+// DrainXferDoneMessages() function drain all message on XferDone channel. There is risk of HandleStatusUpdateManager() already exited and we are blocked.
+// We safeguard against it by checking boolean set on exit of HandleStatusUpdateManager(). We try to make risk factor as low as we can, by calling this function only once.
+//
+// Note: This is not thread safe function. Onus on caller to handle that.
+func (jm *jobMgr) DrainXferDoneMessages() bool {
+	if !jm.jstm.doneHandleStatusUpdateManager && !jm.jstm.xferDoneDrainCalled {
```

Comment:
nit, comment above this

// Poke handleStatusUpdateManager to drain the xferDone channel, but only if it's running.